code/game/machinery/cryo.dm

```diff
@@ -362,41 +381,82 @@ var/global/list/cryo_health_indicator = list(	"full" = image("icon" = 'icons/obj
 	return
 
 /obj/machinery/atmospherics/unary/cryo_cell/update_icon()
-	overlays.len = 0
-	if(on)
-		if(occupant)
-			if(occupant.stat == DEAD || !occupant.has_brain())
-				overlays += cryo_health_indicator["dead"]
-			else
-				if(occupant.health >= occupant.maxHealth)
-					overlays += cryo_health_indicator["full"]
-				else
-					switch((occupant.health / occupant.maxHealth) * 100) // Get a ratio of health to work with
-						if(100 to INFINITY) // No idea how we got here with the check above...
-							overlays += cryo_health_indicator["full"]
-						if(75 to 100)
-							overlays += cryo_health_indicator["good"]
-						if(50 to 75)
-							overlays += cryo_health_indicator["average"]
-						if(25 to 50)
-							overlays += cryo_health_indicator["bad"]
-						if(0 to 25)
-							overlays += cryo_health_indicator["worse"]
-						if(-25 to 0)
-							overlays += cryo_health_indicator["critgood"]
-						if(-50 to -25)
-							overlays += cryo_health_indicator["critaverage"]
-						if(-75 to -50)
-							overlays += cryo_health_indicator["critbad"]
-						if(-100 to -75)
-							overlays += cryo_health_indicator["critworse"]
-						else //Shouldn't ever happen. I really hope it doesn't ever happen.
-							overlays += cryo_health_indicator["dead"]
-			icon_state = "cell-occupied"
-			return
-		icon_state = "cell-on"
+	handle_update_icon()
+
+/obj/machinery/atmospherics/unary/cryo_cell/proc/handle_update_icon() //making another proc to avoid spam in update_icon
+	overlays.Cut()
+	if(!panel_open)
+		icon_state = "pod[on]"
+
+	if(!src.occupant)
+		overlays += "lid[on]" //if no occupant, just put the lid overlay on, and ignore the rest
 		return
-	icon_state = "cell-off"
+
+	if(occupant)
+		var/image/pickle = image(occupant.icon, occupant.icon_state)
+		pickle.overlays = occupant.overlays
+		pickle.pixel_y = 22
+
+		overlays += pickle
+		overlays += "lid[on]"
+		if(src.on && !running_bob_animation) //no bobbing if off
+			var/up = 0 //used to see if we are going up or down, 1 is down, 2 is up
+			spawn(0) // Without this, the icon update will block. The new thread will die once the occupant leaves.
+				running_bob_animation = 1
+				while(occupant)
```

Comment:
oh christ

This will be the source of any lag.  

It won't look bad without it, by the way.  The "cryo chamber" that Luke visited so often in star wars didn't involve any bobbing.